Firmware/ultralcd.cpp

```diff
@@ -8959,6 +8967,43 @@ void lcd_experimental_menu()
     MENU_END();
 }
 
+void reprint_from_eeprom() {
+	char cmd[30];
+	char filename[13];
+	uint8_t depth = 0;
+	char dir_name[9];
+
+	isPrintFinished=false;
+
+	//cmdqueue_reset();
+
+	depth = eeprom_read_byte((uint8_t*)EEPROM_DIR_DEPTH);
+	
+	MYSERIAL.println(int(depth));
+	for (int i = 0; i < depth; i++) {
+		for (int j = 0; j < 8; j++) {
+			dir_name[j] = eeprom_read_byte((uint8_t*)EEPROM_DIRS + j + 8 * i);
+		}
+		dir_name[8] = '\0';
+		MYSERIAL.println(dir_name);
+		// strcpy(dir_names[i], dir_name);
+		card.chdir(dir_name, false);
+	}
+
+	for (int i = 0; i < 8; i++) {
+		filename[i] = eeprom_read_byte((uint8_t*)EEPROM_FILENAME + i);
```

Comment:
you can use a block reading function from the EEPROM, especially when you know the size

Reply:
I'm using the same logic that I have seen when the code is trying to read or write the EEPROM_FILENAME, I suppose that should be easy to recover a block of 8 positions starting from the EEPROM_FILENAME address but I have not tested it.